File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.py
class Solution:
    def shortestBeautifulSubstring(self, s: str, k: int) -> str:
        n = len(s)
        candi = []

        for i in range(n):
            for j in range(i+1, n+1):
                sub = s[i:j]
                if sub.count('1') == k:
                    candi.append(sub)
                    
        if not candi:
            return ""
        candi.sort(key = lambda x: (len(x),x))
        return candi[0]

File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string shortestBeautifulSubstring(string s, int k) {
        int n = s.length();
        string ans = "";
        int min_len = n+1;

        int count = 0;
        int left = 0;

        for (int right = 0; right <n ; ++right){

            if (s[right] == '1'){
                count++;
            }
            while(count == k ){
                while(s[left] == '0'){
                     left++;

                }
            int curr_len = right - left +1;
            string curr_sub = s.substr(left, curr_len);

            if(curr_len < min_len){
                min_len = curr_len;
                ans = curr_sub;

            } else if (curr_len == min_len){
                ans = min(ans, curr_sub);

            }
            if(s[left] == '1'){
                count--;
            }
            left++;

            }
            
        }
        return ans;

    }

};